### ISSUE TITLE
The @mainthread decorator intermittently fails, but only in Pyinstaller created EXE

### ISSUE THREAD

**Issue by DaytonaJohn** [pyinstaller/pyinstaller#3489]
The commented method in the `AThread` class calls a method with the `@mainthread` decorator that opens a `ModalView` popup. This always works correctly when the app is run as `python main.py`, but sometimes fails when the `Pyinstaller` created executable is run. In the failure case, it appears that the decorated method never executes (as evidenced by the log output), and the app hangs on the `self.queue.get(True)` statement. I have tried using `Clock.schedule_once()` instead of the decorator, and different ways of waiting for the popup to be closed, but all behave the same.

To see the error, execute the produced `main.exe` and click on the `Run Test` button, which calls the decorated method. If the error occurs, no popup will appear, and the app will hang. If the error does not occur, a popup will appear with a `Dismiss` button. Clicking the `Dismiss` button will terminate the app. Originally, I designed the app to allow the user to keep cycling through `Run Test` and `Dismiss`, but I discovered that the error only happened on the very first call to the decorated method. If the first call succeeded, then all following calls would also succeed. So I modified the code to make it easier to run it repeatedly, click twice, then run it again. I estimate the failure rate at about 1 in 10 runs. Here is a link to the verbose output of a successful run (a failed run ends at the `waiting in thread: Thread-1` print). I do not have a log for a failure, because the failure never happened when the verbose option was passed to Python (perhaps that is a hint?).

[Successful run log](https://gist.github.com/DaytonaJohn/7f992cb926842df9feb09e608999d7b7)

[Pyinstaller log](https://gist.github.com/DaytonaJohn/7cc96c0a05749309c910a30f9b408017)

Environment:
    Windows 10, 64 bit
    Pyinstaller 3.4.dev0+b31794cf1
    Python v2.7.14:84471935ed
    Kivy v1.10.0

Here is the main.py file:
```
import sys
import threading

from kivy.app import App
from kivy.clock import mainthread
from kivy.lang import Builder
from kivy.uix.button import Button
from kivy.uix.floatlayout import FloatLayout
from kivy.uix.modalview import ModalView

from kivy.compat import PY2
if PY2:
    from Queue import Queue
else:
    from queue import Queue


class TestLayout(FloatLayout):
    def __init__(self):
        super(TestLayout, self).__init__()
        self.th = None

    def do_test(self, *arg):
        self.th = AThread()
        self.th.start()


class MyPopup(object):
    def __init__(self, callback):
        self.buttonCallback = callback
        dismiss_button = Button(text='Dismiss')
        dismiss_button.bind(on_press=self.butt)
        self.popup = ModalView(size_hint=(.5, .5), auto_dismiss=False)
        self.popup.add_widget(dismiss_button)

    def butt(self, *args):
        if self.popup is not None:
            self.popup.dismiss()
        if self.buttonCallback is not None:
            self.buttonCallback()

    @mainthread
    def open(self, *args):
        print('in pop.open in thread: ' + threading.current_thread().name)
        sys.stdout.flush()
        self.popup.open()


class AThread(threading.Thread):
    def __init__(self):
        super(AThread, self).__init__()
        self.daemon = True
        self.pop = None
        self.queue = None

    def button_func(self, *args):
        print('in button_func in thread: ' + threading.current_thread().name)
        self.queue.put(None, False)

    def run(self):
        print('running in thread: ' + threading.current_thread().name)
        sys.stdout.flush()
        self.queue = Queue()
        self.pop = MyPopup(self.button_func)
        print('calling open in thread: ' + threading.current_thread().name)
        sys.stdout.flush()
        self.pop.open()    # This should open the popup, but occasionally it does not
        print('waiting in thread: ' + threading.current_thread().name)
        sys.stdout.flush()
        self.queue.get(True)
        print('done waiting in thread: ' + threading.current_thread().name)
        sys.stdout.flush()
        App.get_running_app().stop()


root = Builder.load_string( '''
TestLayout:
    Button:
        text: 'Run Test'
        on_press: root.do_test()
''')


class testApp(App):
    def build(self):
        return root


if __name__ == "__main__":
    testApp().run()
```

And here is the main.spec file:
```
# -*- mode: python -*-

block_cipher = None

a = Analysis(['main.py'],
             pathex=['C:\\Users\\John\\PycharmProjects\\popupbug'],
             binaries=[],
             datas=[],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          exclude_binaries=True,
          name='main',
          debug=False,
          strip=False,
          upx=False,
          console=True )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=False,
               upx=True,
               name='main')
```

**Comment by DaytonaJohn** [comment]
Found a better fix/workaround. If I put

`from kivy.config import Config
Config.set('kivy', 'kivy_clock', 'interrupt')`

at the top of my `main.py`, then the problem goes away. I did a bit of tracing and found that the main thread was hanging on a sleep method in kivy.clock. The above code changes the clock that kivy uses. Just tried changing it, hoping that a different clock might not show the problem, and got lucky.
